app/gui2/src/components/CreateNewNode.vue

```diff
@@ -0,0 +1,104 @@
+<script setup lang="ts">
```

Comment:
This component name is a bit generic--we have other buttons that create new nodes too.
Since this component doesn't define much HTML or CSS, you might consider encapsulating using a `.ts` module to achieve this logic encapsulation.